sdks/python/apache_beam/transforms/util.py

```diff
@@ -518,10 +563,13 @@ def expand(self, pcoll):
       key_coder = coders.registry.get_coder(typehints.Any)
       value_coder = key_coder
 
+    gbk = beam.GroupByKey()
+    gbk._inside_gbek = True
+
     return (
         pcoll
         | beam.ParDo(_EncryptMessage(self._hmac_key, key_coder, value_coder))
-        | beam.GroupByKey()
+        | gbk
```

Comment:
I guess this breaks the update compatibility given the line number changes.

Reply:
That's true - that's why I think we should do https://github.com/apache/beam/pull/36251#discussion_r2380200051 to permanently avoid this problem.

I can follow up with a PR

Reply:
https://github.com/apache/beam/pull/36381 will fix this broadly.